nilearn/plotting/surface/_plotly_backend.py

```diff
@@ -353,3 +353,21 @@ def _plot_surf(
             plotly_figure.savefig()
 
         return plotly_figure
+
+    def _plot_surf_contours(
+        self,
+        surf_mesh=None,
+        roi_map=None,
+        hemi=None,
+        levels=None,
+        labels=None,
+        colors=None,
+        legend=False,
+        cmap="tab20",
+        title=None,
+        output_file=None,
+        axes=None,
+        figure=None,
+        **kwargs,
+    ):
+        raise NotImplementedError()
```

Comment:
Shall we implement this for `plotly` as well?

Reply:
nope 
pretty sure that for plotly we rely on the figure.add_contours method to do this

so maybe making sure that the error message mention this would be better

Reply:
See 

https://github.com/nilearn/nilearn/blob/34ff77ac402c09a6f49284359defa6aa4de9b07f/examples/01_plotting/plot_3d_map_to_surface_projection.py#L179-L207

Reply:
Wouldn't it be better to adopt the same behavior for both engines?

I would suggest to add `engine` parameter to `plot_surf_contours` and implement it for `plotly`. As the implementation already exists, it will be easy I think.

In the long term, I would suggest to create a `matplotlib` equivalent of `PlotlySurfaceFigure`, ie. `MatplotlibSurfaceFigure` that has the same functionality. What do you think? 

It seems a bit strange to me that depending on the engine we have different objects and different way of doing things. I think from users point of view, it should be enough to know the interface, and use interface methods/or functions for all engines. The implementation details should not be an issue for the user.